File: C#/ASCII Snake/ASCII Snake/Arena.cs
using System;
using System.Text;

namespace ASCII_Snake
{
    public class Arena
    {
        public const string __wallStr = "# ";
        public const string __snakeStr = "S ";
        public const string __foodStr = "F ";
        public const string __noneStr = "  ";

        public readonly int _width;
        public readonly int _height;

        string[,] _arena;

        public string[,] GetArena => _arena;

        public Arena(int width, int height)
        {
            _width = width;
            _height = height;
            _arena = new string[_width, _height];
            for (int h = 0; h < _height; h++)
            {
                for (int w = 0; w < _width; w++)
                {
                    if(w == 0 || w == _width - 1 || h == 0 || h == _height - 1)
                    {
                        _arena[w, h] = __wallStr;
                    }
                    else
                    {
                        _arena[w, h] = __noneStr;
                    }
                }
            }
        }

        public void SetPosition(Vector2Int position, string value)
        {
            _arena[position.GetX, position.GetY] = value;
        }

        public void RemovePosition(Vector2Int position)
        {
            _arena[position.GetX, position.GetY] = __noneStr;
        }

        public string GetStringAsListIndex(int index, out Vector2Int position)
        {
            int row = index / _width;
            int col = index - row * _width;

            position = new Vector2Int(row, col);
            return _arena[row, col];
        }

        public string Print()
        {
            StringBuilder sb = new StringBuilder();
            for (int h = 0; h < _height; h++)
            {
                for (int w = 0; w < _width; w++)
                {
                    sb.Append(_arena[w, h]);
                }

                sb.Append("\n");
            }

            return sb.ToString();
        }
    }
}

File: C++/ASCII Snake/ASCII Snake/Arena.cpp
#include "Arena.h"

Arena::Arena(int width, int height)
{
	this->width = width;
	this->height = height;

	for (size_t h = 0; h < height; h++)
	{
		std::vector<std::string> row;
		for (size_t w = 0; w < width; w++)
		{
			if (h == 0 || w == 0 || h == height - 1 || w == width - 1) {
				row.push_back(__wallStr);
			}
			else
			{
				row.push_back(__noneStr);
			}
		}

		arena.push_back(row);
	}
}

Arena::~Arena()
{
}

void Arena::setString(Vector2Int position, std::string value)
{
	arena[position.x][position.y] = value;
}

Vector2Int Arena::indexToVector2Int(int index)
{
	int x = index / width;
	int y = index - x * width;
	return Vector2Int(x, y);
}

std::string Arena::getString(Vector2Int position)
{
	return arena[position.x][position.y];
}

std::string Arena::toString()
{
	std::string result = "";
	for (size_t h = 0; h < height; h++)
	{
		for (size_t w = 0; w < width; w++)
		{
			result += arena[w][h];
		}

		result += "\n";
	}

	return result;
}
